adam/visualization/make_scenes.py

```diff
@@ -0,0 +1,156 @@
+from typing import (
+    AbstractSet,
+    Any,
+    Callable,
+    Iterable,
+    List,
+    Tuple,
+    TypeVar,
+    Union,
+    Optional,
+)
+
+import random
+from collections import defaultdict
+from attr import attrib, attrs
+from attr.validators import instance_of
+
+from vistautils.parameters import Parameters
+from vistautils.parameters_only_entrypoint import parameters_only_entry_point
+
+from immutablecollections import ImmutableListMultiDict, immutablelistmultidict
+
+
+from adam.language.dependency import LinearizedDependencyTree
+
+from adam.curriculum.phase1_curriculum import GAILA_PHASE_1_CURRICULUM
+from adam.experiment import InstanceGroup
+from adam.geon import CrossSection
+
+from adam.situation.high_level_semantics_situation import HighLevelSemanticsSituation
+from adam.perception.developmental_primitive_perception import (
+    DevelopmentalPrimitivePerceptionFrame,
+    RgbColorPerception,
+)
+
+from adam.visualization.panda3d_interface import SituationVisualizer
+from adam.visualization.utils import Shape
+
+USAGE_MESSAGE = """ """
+
+
+def main() -> None:
+    random.seed(2015)
+
+    # go through curriculum scenes (fed in from where?) and output geometry types
+    print("scene generation test")
+    sc = SceneCreator()
+    viz = SituationVisualizer()
+    for i, scene in enumerate(sc.create_scenes(GAILA_PHASE_1_CURRICULUM)):
+        print(f"SCENE {i}")
+        for obj, properties in scene.items():
+            if obj.geon is None:  # only interested in rendering geons
+                continue
+            # print(obj.geon.cross_section_size)
+            # print(obj.axes)
+            obj_added = False  # TODO: clean up this bit of control-flow mess
+            for prop in properties:
```

Comment:
nit: Cleaner would be to loop over the properties and setting a `color` variable which is initially `None`.  Then you can just have one call to `add_model`.